apps/user-service/src/auth/reset-password.controller.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Controller,
+  Body,
+  Post,
+  HttpStatus,
+  BadRequestException,
+  Param, Put
+} from "@nestjs/common";
+import { ResetPasswordService } from './reset-password.service';
+import { ApiOperation } from '@nestjs/swagger';
+import { JsonApiResponse } from '@terramatch-microservices/common/decorators';
+import { buildJsonApi, JsonApiDocument } from '@terramatch-microservices/common/util';
+import { ApiException } from '@nanogiants/nestjs-swagger-api-exception-decorator';
+import { ResetPasswordRequest } from "./dto/reset-password-request.dto";
+import { ResetPasswordDto } from './dto/reset-password.dto';
+import { NoBearerAuth } from '@terramatch-microservices/common/guards';
+import { ResetPasswordResponseDto } from "./dto/reset-password-response.dto";
+
+@Controller('auth/v3/passwordResets')
+export class ResetPasswordController {
+  constructor(private readonly resetPasswordService: ResetPasswordService) {}
+
+  @Post()
+  @NoBearerAuth
+  @ApiOperation({
+    operationId: 'requestPasswordReset',
+    description: 'Send password reset email with a token',
+  })
+  @JsonApiResponse({ status: HttpStatus.CREATED, data: { type: ResetPasswordResponseDto } })
+  @ApiException(() => BadRequestException, { description: 'Invalid request or email.' })
+  async requestReset(@Body() { emailAddress, callbackUrl }: ResetPasswordRequest): Promise<JsonApiDocument> {
+    const { email, userId, uuid } = await this.resetPasswordService.sendResetPasswordEmail(emailAddress, callbackUrl);
+    return buildJsonApi()
+      .addData(`${userId}`, new ResetPasswordResponseDto({ emailAddress: email, uuid }))
```

Comment:
Why is both the user id and the user UUID needed? Maybe I'll see it when I look at the FE integration, but I suspect this would do better putting the UUID in as the resource ID and leaving the user ID out of it.